### PR TITLE
fix: improve llm parsing

### LINKED ISSUE #795: [BUG]: Wrong parsing of LLM answers
### Describe the bug

After asking an LLM for a question, even though it answers the question properly, they're parsed wrong. 

### Steps to reproduce

This happens in [find_best_match](https://github.com/AIHawk-co/Auto_Jobs_Applier/blob/94b2a7648706898f7da87bf6e42cff1d734d05ce/src/ai_hawk/llm/llm_manager.py#L370) function. It tries to find the best match from the options based on the answer of the LLM. 

## Logs 
Screenshot 2024-11-09 at 8 36 23 PM

In this case, the answer was `no`, but when finding the best match from the options, it determined `yes` which is wrong. 

### Expected behavior

Parse answers properly

### Actual behavior

Sometimes LLM responses are not parsed correctly. 

### Branch

None

### Branch name

_No response_

### Python version

_No response_

### LLM Used

Gemini

### Model used

Gemini-1.5 flash

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by 49Simon**
Fixes #795
- Force the llm to output the answer only
- Clean markdown output, if any
- This will improve the distance calculation of Levenshtein, when using the `find_best_match` function

### CODE REVIEW

File: src/ai_hawk/llm/llm_manager.py
Comment (surapuramakhil) [inline src/ai_hawk/llm/llm_manager.py:576]: I would suggest rather than cleanig forcing LLM is better option. (prompt engineering)

File: src/ai_hawk/llm/llm_manager.py
Comment (surapuramakhil) [inline src/ai_hawk/llm/llm_manager.py:596]: I believe sometimes these questions not only be one word answer questions.

something like "why are  you interested in <compnay name>?"

* serve as bullet points

**Review by surapuramakhil — CHANGES_REQUESTED** [review]
I am slightly skeptical about LLM output cleaning. removing # and * from output.

No issue in updating prompts for forcing LLMs.

@49Simon request to have second review by someone who previously contributed to AI aspect to this project.

**Comment by feder-cr** [comment]
I will @surapuramakhil

**Comment by 49Simon** [comment]
@feder-cr let me know what you think of this. Will tag @cjbbb and @sarob if they've any input on this

**Review by feder-cr — APPROVED** [review]

**Review by surapuramakhil — APPROVED** [review]

**Pull request merged**
